algosdk/atomic_transaction_composer.py

```diff
@@ -59,7 +59,7 @@ def populate_foreign_array(
     if zero_value and value_to_add == zero_value:
```

Comment:
```suggestion
    if zero_value is not None and value_to_add == zero_value:
```

Reply:
Yeah this looks good, didn't notice that. Its usually a good idea to specifically check for None and 0 values.